pkg/action/executor/pod_quota_evaluator.go

```diff
@@ -0,0 +1,487 @@
+/*
+Copied from k8s.io/kubernetes/pkg/quota/v1/evaluator/core/pods.go
+*/
+
+package executor
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation"
+	quota "k8s.io/apiserver/pkg/quota/v1"
+	"k8s.io/apiserver/pkg/quota/v1/generic"
+	"k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+)
+
+// the name used for object count quota
+var podObjectCountName = generic.ObjectCountQuotaResourceNameFor(corev1.SchemeGroupVersion.WithResource("pods").GroupResource())
+
+const (
+	// Allow specifying NamespaceSelector in PodAffinityTerm.
+	PodAffinityNamespaceSelector featuregate.Feature = "PodAffinityNamespaceSelector"
```

Comment:
It doesn't make sense to have these featuregates and the code relevant to them here without we enabling them. It will never be hit.
Now `PodAffinityNamespaceSelector` is beta and is [enabled by default in k8s 1.22](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#namespace-selector). I think we can include the code without the feature gate check.
Same with `PodAffinityNamespaceSelector` [it was beta in 1.18](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-overhead/) and more relevant to `pod.Spec.Overhead` field present in the pods spec. I think we can include the code sans feature gate.

Reply:
@irfanurrehman  I removed those code in the commit 5df5f8d6fae8b0425b17eab2251c29ebb0e60346

Reply:
@irfanurrehman , please have a look at my new commit https://github.com/turbonomic/kubeturbo/pull/658/commits/e1e72acab9c4af49efee5bff8213ab844694c1ce